src/Database.php

```diff
@@ -280,9 +280,9 @@ public function createCollection($collectionName, array $options = [])
             $options['typeMap'] = $this->typeMap;
         }
 
-        $server = select_server($this->manager, new ReadPreference(ReadPreference::RP_PRIMARY), extract_session_from_options($options));
+        $server = select_server($this->manager, $options);
 
-        if (! isset($options['writeConcern']) && server_supports_feature($server, self::$wireVersionForWritableCommandWriteConcern)) {
+        if (! isset($options['writeConcern']) && server_supports_feature($server, self::$wireVersionForWritableCommandWriteConcern) && ! is_in_transaction($options)) {
```

Comment:
IIRC, these checks were not originally added in the PR for [PHPLIB-368](https://jira.mongodb.org/browse/PHPLIB-368) because the operations themselves weren't supported within a transaction. I think this makes sense, though, as it makes it less likely we'll get a client-side error for providing an explicit RC/WC (be it from our driver or libmongoc) and more likely the user will get a server-side error for running an unsupported command within a transaction.

I suppose this is similar in some respects to your earlier PR to consider the pinned session when resuming a change stream.

Reply:
Yep, I thought it best to add it for consistency, and to avoid failures down the line when we realise that transactions allow more operations than I envisioned. I'm not sure if 4.4 will support DDL commands in transactions, but this will have us prepared either way.